docs/concepts/clustering.rst

```diff
@@ -138,48 +137,50 @@ the next update.
 Master Node Election
 --------------------
 
-The process of electing a node as the master node in a cluster is called Master
-Node Election. There must be only one master node per cluster at any single
-time. In a CrateDB cluster any node is eligible to be elected as a master node,
-although this could also be restricted to a subset of nodes if required. These
-nodes will then elect a single node as master to coordinate the cluster state
-across the cluster.
-
+In a CrateDB cluster there can only be one master node at any single time.
+The cluster only becomes available to serve requests once a master has been 
+elected, and a new election takes place if the current master node becomes 
+unavailable.
+By default all nodes are master-eligible, but 
+:ref:`a node setting <node.master>`
+is available to indicate, if desired, that a node must not take on the role
+of master. To elect a master among the eligible nodes, a majority 
+(``floor(half)+1``), also known as *quorum*, is required among a subset of 
+all master-eligible nodes, this subset of nodes is known as the
+*voting configuration*.
+The *voting configuration* is a list which is persisted as part of the cluster
+state. It is maintained automatically in a way that makes so that split-brain
+scenarios are never possible.
+Every time a node joins the cluster, or leaves the cluster, even if it is 
+for a few seconds, CrateDB re-evaluates the voting configuration.
+If the new number of master-eligible nodes in the cluster is odd, CrateDB will
+put them all in the voting configuration.
+If the number is even, CrateDB will exclude one of the master-eligible nodes
+from the voting configuration.
+A master node rescinds its role if it cannot contact a quorum of nodes from
+the latest voting configuration.
+
+.. WARNING::
+
+   If you do infrastructure maintenance, please note that as nodes are shutdown 
```

Comment:
Imho, it would be nice to include info from the former version, to show how a split-brain situation is avoided, when the network is split in 3-2, 3-1-1, 2-2-1, 1-1-1 examples.

Reply:
The explanation on how the voting configuration works seems quite complex already, and the warning note serves as an example. I worry that adding more examples may create confusion and I have not found any "formal" explanation on how this logic can cover all possible network partition scenarios, but in my tests it really seems to work.
Do you have any particular example in mind?

Reply:
Maybe it would be nice to mention at least one example, that if there is a network partition, where 3 nodes form a cluster and the other 2 nodes a 2nd cluster, only the 1st one can elect a master and continue to operate (change data: insert/update/delete) and after the network partition is resolved, the cluster has remained consistent, since no write operation could have been performed in the 2nd (2-node) cluster.

Reply:
My tests always focused on starting from cold and could not get the minority partition to start and serve read requests, but I will try again partition the cluster on the fly with the nodes running.

Reply:
We spoke with @proddata about these network partition tests, we want to do these, but they are not trivial and we do not want to hold back the many clarifications in this PR and a tutorial that we have drafted for the Community which references the updated documentation, so we will address this as a separate task (already in our list).